android/src/main/java/io/github/reactnativecommunity/javascriptcore/JSCExecutorFactory.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package io.github.reactnativecommunity.javascriptcore
+
+import com.facebook.react.bridge.JavaScriptExecutor
+import com.facebook.react.bridge.JavaScriptExecutorFactory
+import com.facebook.react.bridge.WritableNativeMap
+
+@Suppress("Unused")
+class JSCExecutorFactory(private val appName: String, private val deviceName: String) :
+    JavaScriptExecutorFactory {
+
+  @Throws(Exception::class)
+  override fun create(): JavaScriptExecutor {
+    val jscConfig =
+        WritableNativeMap().apply {
+          putString("OwnerIdentity", "ReactNative")
+          putString("AppIdentity", appName)
+          putString("DeviceIdentity", deviceName)
+        }
+    return JSCExecutor(jscConfig)
+  }
+
+  override fun startSamplingProfiler() {
+    throw UnsupportedOperationException("Starting sampling profiler not supported on ${toString()}")
+  }
+
+  override fun stopSamplingProfiler(filename: String) {
+    throw UnsupportedOperationException("Stopping sampling profiler not supported on ${toString()}")
+  }
+
+  override fun toString(): String = "JSIExecutor+JSCRuntime"
```

Comment:
is this right?

Reply:
just copy from core: https://github.com/facebook/react-native/blob/1b066bac9abff457b8baf85a21ca255cf103abfe/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/jscexecutor/JSCExecutorFactory.kt#L36

that's an old arch convention. there's also `JSIExecutor+HermesRuntime`. i don't want to change this for community jsc. let's keep it as-is.

Reply:
Ah yes, let's keep ti